### PR TITLE
CI: Add HPSF GitLab CI file

### PR DESCRIPTION
**Pull request by masterleinad**
This currently works correctly on my GitLab mirror (https://gitlab.spack.io/masterleinad/kokkos/-/jobs/12714093) using https://hub.docker.com/layers/masterleinad/kokkos-nvcc/12.6.1/images/sha256-fdae939e761af9d1d0242c562c92218e497f9c33bfe7b5e664b746d365e076f7?context=explore (which is basically just https://hub.docker.com/layers/nvidia/cuda/12.6.1-devel-ubuntu24.04/images/sha256-16696cc4610b50d57d8e9f5b9f776bca4e17ceb800084e7f009c42bd72a3a771?context=explore with `python3` and `cmake`.

### CODE REVIEW

File: .hpsf-gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+test:
+  stage: test
+  tags: [nvidia-gh200]
+  image: masterleinad/kokkos-nvcc:12.6.1
```
Comment (masterleinad) [inline .hpsf-gitlab-ci.yml:4]: Not sure if we want to change to something more "official".

Comment (aprokop) [reply .hpsf-gitlab-ci.yml:4]: Probably. Would also need a recipe how to build it.

Comment (masterleinad) [reply .hpsf-gitlab-ci.yml:4]:
```
FROM nvidia/cuda:12.6.1-devel-ubuntu24.04

RUN apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub

RUN apt-get update && apt-get install -y \
        bc \
        wget \
        python3 \ 
        cmake \
        && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
```

Comment (masterleinad) [reply .hpsf-gitlab-ci.yml:4]: What about we find a better solution later so we can continue with automating the GitLab integration?

**Review by aprokop — APPROVED** [review]

**Review by cedricchevalier19 — APPROVED** [review]
I would like to know if we want to put this file and probably the future files for gitlab-ci in a specific directory.

**Comment by masterleinad** [comment]
> I would like to know if we want to put this file and probably the future files for gitlab-ci in a specific directory.

Good idea! I created a `.gitlab`  directory and moved the file there.

File: .hpsf-gitlab-ci.yml (hunk 1 above)
Comment (dalg24) [reply .hpsf-gitlab-ci.yml]: Sure.  Any particular reason why you wrote your own dockerfile rather than reuse `scripts/docker/Dockerfile.nvcc`?

**Comment by dalg24** [comment]
> > I would like to know if we want to put this file and probably the future files for gitlab-ci in a specific directory.
> 
> Good idea! I created a `.gitlab` directory and moved the file there.

@Rombur maybe we can move `.olcf-gitlab-ci.yml -> .gitlab/olcf-gitlab-ci.yml` too

File: .hpsf-gitlab-ci.yml (hunk 1 above)
Comment (masterleinad) [reply .hpsf-gitlab-ci.yml]: I wasn't able to run the `cmake` installed that way
> /usr/bin/bash: line 156: /opt/cmake/bin/cmake: cannot execute binary file: Exec format error

Also I needed to install `python3`, see https://gitlab.spack.io/masterleinad/kokkos/-/jobs/12702052 (which we could of course also add to Dockerfile.nvcc).

Comment (dalg24) [reply .hpsf-gitlab-ci.yml]: There is an ADDITIONAL_PACKAGES argument

**Pull request merged**